src/credits.js

```diff
@@ -0,0 +1,164 @@
+  it: 'it-IT',
+  lv: 'lv-LV',
+  nl: 'nl-NL',
+  pl: 'pl-PL',
+  ro: 'ro-RO',
+  sl: 'sl-SI',
+  sv: 'sv-SE'
+};
+
+const fetchEssence = async(type, id) => {
+  let sql;
+
+  switch (type) {
+    case 'Alchemy::EssenceText':
+      sql = `select body as value from alchemy_essence_texts where id=${id}`;
+      break;
+    case 'Alchemy::EssencePicture':
+      sql = `
+        select ap.image_file_uid as value
+        from alchemy_essence_pictures aep
+        inner join alchemy_pictures ap on aep.picture_id=ap.id
+        where aep.id=${id}
+      `;
+      break;
+    case 'Alchemy::EssenceRichtext':
+      sql = `select body as value from alchemy_essence_richtexts where id=${id}`;
+      break;
+    default:
+      throw new Error(`Unknown type ${type}`);
+  }
+
+  const res = await pgClient.query(sql);
+  if (res.rows.length === 0) return null;
+
+  const row = res.rows[0];
+  if (!row.value || row.value === '') return null;
+
+  row['type'] = type;
+
+  return row;
+};
+
+const contentfulAssetForAlchemyPicture = async(imageFileUid) => {
+  const uid = encodeURIComponent(imageFileUid);
+  const assetId = imageLog[uid];
+  if (!assetId) return '';
+
+  const asset = await contentfulConnection.getAsset(assetId);
+
+  return turndownService.turndown(`<img src="https:${asset.fields.file['en-GB'].url}"/>`);
+};
+
+const creditsFromEssences = async(essences) => {
+  let credits = '';
+  for (const essence of essences) {
+    switch (essence.type) {
+      case 'Alchemy::EssenceText':
+        credits = credits + `## ${essence.value}\n`;
+        break;
+      case 'Alchemy::EssencePicture':
+        credits = credits + await contentfulAssetForAlchemyPicture(essence.value);
+        break;
+      case 'Alchemy::EssenceRichtext':
+        credits = credits + await turndownService.turndown(essence.value);
+        break;
+      default:
+        throw new Error(`Unknown type ${essence.type}`);
+    }
+  }
+  return credits;
+};
+
+const creditsFromRow = async(rowEssences) => {
+  const essences = [];
+  for (const essence of rowEssences) {
+    const essenceData = await fetchEssence(essence.type, essence.id);
+    if (essenceData) essences.push(essenceData);
+  }
+  return await creditsFromEssences(essences);
+};
+
+const creditExhibition = async(urlname, rows) => {
+  console.log(urlname);
+  const exhibitionSlug = urlname.split('/')[0];
+
+  const entries = await contentfulConnection.getEntries({
+    'content_type': 'exhibitionPage',
+    'locale': 'en-GB',
+    'fields.identifier': exhibitionSlug,
+    'limit': 1
+  });
+  const entry = entries.items[0];
+
+  if (!entry.fields.credits) entry.fields.credits = {};
+
+  for (const locale in rows) {
+    const contentfulLocale = localeMap[locale];
+    console.log(`- ${locale} => ${contentfulLocale}`);
+    const credits = await creditsFromRow(rows[locale]);
+    entry.fields.credits[contentfulLocale] = credits;
+  }
+
+  const updated = await entry.update();
+  await updated.publish();
```

Comment:
when the entry isn't publishable this can cause the script to hang. 
Consider catching any errors here.